skyportal/handlers/api/user_obj_list.py

```diff
@@ -0,0 +1,366 @@
+        ):
+            return self.error('Insufficient permissions to access this listing. ')
+
+        obj_id = data.get('obj_id')
+        if Obj.query.get(obj_id) is None:  # verify that object exists!
+            return self.error(f'Object "{obj_id}" does not exist!')
+
+        list_name = data.get('list_name')
+        if not check_list_name(list_name):
+            return self.error(
+                "Input `list_name` must begin with alphanumeric/underscore"
+            )
+
+        query = (
+            DBSession()
+            .query(Listing)
+            .filter(
+                Listing.user_id == user_id,
+                Listing.obj_id == obj_id,
+                Listing.list_name == list_name,
+            )
+        )
+
+        # what to do if listing already exists...
+        if query.first() is not None:
+            if error_if_exists:
+                return self.error(
+                    f'Listing already exists with user_id={user_id}, obj_id={obj_id} and list_name={list_name}'
+                )
+            else:  # if user, obj and list name are identical, no need for an update of existing record
+                return self.success(data={'listing_id': query.first().id})
+
+        # no such listing, can just add a new one!
+        listing = Listing(user_id=user_id, obj_id=obj_id, list_name=list_name)
+
+        DBSession().add(listing)
+        DBSession().commit()
+
+        return self.success(data={'listing_id': listing.id})
+
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Add a listing.
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  user_id:
+                    type: integer
+                    required: true
+                  obj_id:
+                    type: string
+                    required: true
+                  list_name:
+                    type: string
+                    required: true
+                    description: |
+                        Listing name for this item, e.g., "favorites".
+                        Multiple objects can be saved by the same user to different
+                        lists, where the list names are user-defined.
+                        List name must be a non-empty string starting with an
+                        alphanumeric character or underscore.
+                        (it must match the regex: /^\\w+/)
+
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            listing_id:
```

Comment:
```suggestion
                            id:
```
For consistency with other handlers